### DISCUSSION TITLE
FOCAL programming language

### DISCUSSION THREAD

**Discussion by btiffin** [s390guy/vm370#30] [Ideas]
Hello, Bob,

Hope I'm not using this Discussion subtab irresponsibly.  A) I wanted to check out GitHub discussions, just to see and B) the reason for the note.

Would you like to add a FOCAL implementation to VMCE, Bob?

https://en.wikipedia.org/wiki/FOCAL_%28programming_language%29

Dave Pitts, who contributed the code for the -t and -T program listing support in GnuCOBOL, also wrote a FOCAL implementation back in the day (1978ish, updated as recently as 2011).  GPLv2 licensed.

It was one of my first attempts at C in VM/370, and I just amalgamated all the sources into a single `focalcms.c` for trials.  Works the charm.  (Little testing so far, but focalcms runs most FOCAL sources I've found in the wild so far). 

If you think it would worthwhile, then drop hints on how you like things packaged.  I'm still utterly green to the ways of VM/370 under Hercules.  I've written a FOCALCMS HELPCMD and FOCALCMS HELPCMD2 just because, and they may need fleshing out to a 'release quality'.

`focalcms` supports some of the `LIBRARY subcommand` extensions, and I think it'd be neat to add a `LIBRARY SYSTEM cmscommand` (or somesuch).

For others to try, I've been plopping these files at http://btiffin.users.sourceforge.net/files/

If you think it might be worthwhile. I'll clean it up for packaging and releasing.

**Comment by BobBolch** [comment]
Hi Brian,
I think this is very worthwhile! I took a look at your development area,
and I think you have everything we need. I will have some free time in
about a week, to
download it from your directory and play with it a little bit. When you
have it just like
you want, I will add it to the distribution plan. Thank you so much for
thinking to
make this available for VMers.
Bob

**Reply by btiffin** [reply]
Cool.  If you like what you see, go ahead and add it in, as is, Bob.  I'll drop Dave a note so he knows his code is getting a new potential audience.  ;-)
